src/vma/dev/net_device_val.cpp

```diff
@@ -1567,7 +1567,7 @@ bool net_device_val::verify_enable_ipoib(const char* interface_name)
 #ifndef DEFINED_IBV_QP_INIT_SOURCE_QPN
 	// Dont offload mlx5 devices if source qpn is not defined.
 	ib_ctx_handler* ib_ctx = g_p_ib_ctx_handler_collection->get_ib_ctx(get_ifname_link());
-	if (!strncmp(ib_ctx->get_ibname(), "mlx5", 4)) {
+	if (!ib_ctx->is_mlx4()) {
```

Comment:
It is dangerous way to check. Let imagine that there is more than just mlx4 and mlx5. Please consider `is_mlx5()` to be consistent.